--- plg2_s_signature_mod.c	2022-06-22 12:26:03.940567986 +0200
+++ project_libgit2/src/signature.c	2021-09-27 22:09:50.000000000 +0200
@@ -168,8 +168,7 @@
 	time(&now);
 	utc_tm = p_gmtime_r(&now, &_utc);
 	utc_tm->tm_isdst = -1;
-    offset = mktime(utc_tm) - now;
-//	offset = (time_t)difftime(now, mktime(utc_tm));
+	offset = (time_t)difftime(now, mktime(utc_tm));
 	offset /= 60;
 
 	if (git_signature_new(&sig, name, email, now, (int)offset) < 0)
